src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -695,12 +759,32 @@ public int compare(Description o1, Description o2) {
     return filteredRequests;
   }
 
+  @VisibleForTesting
+  void runShutdownHooks() throws InterruptedException {
+    for(Thread hook: shutdownHooks) {
+      hook.start();
+    }
+    for(Thread hook: shutdownHooks) {
+      hook.join(1000); // wait for all hooks to complete, up to 1 second
```

Comment:
This feels short enough to be flaky. Would be good to extend to account for busy hosts.